examples/network-hhmodel/seir.rs

```diff
@@ -0,0 +1,191 @@
+            context.set_person_property(e.neighbor, DiseaseStatus, DiseaseStatusValue::E);
+            println!("Person {} exposed person {}.", e.person, e.neighbor);
+        }
+    }
+}
+
+fn schedule_waiting_event(
+    context: &mut Context,
+    person_id: PersonId,
+    shape: f64,
+    mean_period: f64,
+    new_status: DiseaseStatusValue,
+) {
+    let ct = context.get_current_time();
+    let waiting_time = calculate_waiting_time(context, shape, mean_period);
+
+    context.add_plan(ct + waiting_time, move |context| {
+        context.set_person_property(person_id, DiseaseStatus, new_status);
+    });
+}
+
+fn schedule_infection(context: &mut Context, person_id: PersonId) {
+    let parameters = context
+        .get_global_property_value(Parameters)
+        .unwrap()
+        .clone();
+
+    schedule_waiting_event(
+        context,
+        person_id,
+        parameters.shape,
+        parameters.incubation_period,
+        DiseaseStatusValue::I,
+    );
+}
+
+fn schedule_recovery(context: &mut Context, person_id: PersonId) {
+    let parameters = context
+        .get_global_property_value(Parameters)
+        .unwrap()
+        .clone();
+
+    schedule_waiting_event(
+        context,
+        person_id,
+        parameters.shape,
+        parameters.infectious_period,
+        DiseaseStatusValue::R,
+    );
+}
+
+pub fn init(context: &mut Context, initial_infections: Vec<PersonId>) {
+    context.add_periodic_plan_with_phase(
+        1.0,
+        |context| {
+            println!("Current time is {}.", context.get_current_time());
+
+            let parameters = context
+                .get_global_property_value(Parameters)
+                .unwrap()
+                .clone();
+
+            // infect the networks
+            expose_network::<Household>(
+                context,
+                sar_to_beta(parameters.sar, parameters.incubation_period),
+            );
+            expose_network::<AgeUnder5>(
+                context,
+                sar_to_beta(
+                    parameters.sar / parameters.between_hh_transmission_reduction,
+                    parameters.incubation_period,
+                ),
+            );
+            expose_network::<Age5to17>(
+                context,
+                sar_to_beta(
+                    parameters.sar / parameters.between_hh_transmission_reduction,
+                    parameters.incubation_period,
+                ),
+            );
+        },
+        ExecutionPhase::Normal,
+    );
+
+    context.subscribe_to_event(
+        move |context, event: PersonPropertyChangeEvent<DiseaseStatus>| {
+            match event.current {
+                DiseaseStatusValue::E => schedule_infection(context, event.person_id),
+                DiseaseStatusValue::I => schedule_recovery(context, event.person_id),
+                _ => (),
+            };
+        },
+    );
+
+    // expose the first person to the disease
```

Comment:
```suggestion
    // expose the first people to the disease
```

It's true we only start with one person, but that's not part of the API guarantee of this function.